transport/transport.go

```diff
@@ -168,6 +168,9 @@ type Stream struct {
 	id uint32
 	// nil for client side Stream.
 	st ServerTransport
+	// Keep the user context for stats handling.
+	// All stats handling should use the user context instead of the stream context.
```

Comment:
All stats collection should use the user context (instead of the stream context) so that all the generated stats for a particular RPC can be associated in the processing phase.